DependencyInjection/Compiler/SerializerConfigurationPass.php

```diff
@@ -41,31 +41,26 @@ public function process(ContainerBuilder $container)
             throw new \InvalidArgumentException('Neither a service called "jms_serializer.serializer" nor "serializer" is available and no serializer is explicitly configured. You must either enable the JMSSerializerBundle, enable the FrameworkBundle serializer or configure a custom serializer.');
         }
 
-        if ($container->has('serializer')) {
-            $class = $container->getParameterBag()->resolveValue(
-                $container->findDefinition('serializer')->getClass()
-            );
-
-            if (is_subclass_of($class, 'Symfony\Component\Serializer\SerializerInterface')) {
-                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.symfony');
-                $container->removeDefinition('fos_rest.serializer.exception_wrapper_serialize_handler');
-            } elseif (is_subclass_of($class, 'JMS\Serializer\SerializerInterface')) {
-                $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
-            } elseif (is_subclass_of($class, 'FOS\RestBundle\Serializer\Serializer')) {
-                $container->setAlias('fos_rest.serializer', 'serializer');
-            } else {
-                throw new \InvalidArgumentException(sprintf('"fos_rest.serializer" must implement FOS\RestBundle\Serializer\Serializer (instance of "%s" given).', $class));
-            }
+        if ($container->has('jms_serializer.serializer')) {
+            $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
 
             return;
-        } else {
-            $container->removeDefinition('fos_rest.serializer.exception_wrapper_normalizer');
         }
 
-        if ($container->has('jms_serializer.serializer')) {
-            $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.jms');
+        // Remove the exception normalizer linked to the jms normalizer
+        $container->removeDefinition('fos_rest.serializer.exception_normalizer.jms');
+
+        // As there is no `jms_serializer.serializer` service, there is a `serializer` service
+        $class = $container->getParameterBag()->resolveValue(
+            $container->findDefinition('serializer')->getClass()
+        );
+
+        if (is_subclass_of($class, 'Symfony\Component\Serializer\SerializerInterface')) {
+            $container->setAlias('fos_rest.serializer', 'fos_rest.serializer.symfony');
+        } elseif (is_subclass_of($class, 'FOS\RestBundle\Serializer\Serializer')) {
+            $container->setAlias('fos_rest.serializer', 'serializer');
```

Comment:
I don't think it has ever been supported, what's supported is [removing the `serializer` alias](https://github.com/schmittjoh/JMSSerializerBundle/blob/master/DependencyInjection/Configuration.php#L45).